src/proficiency.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#ifndef CATA_SRC_PROFICIENCY_H
+#define CATA_SRC_PROFICIENCY_H
+
+#include "generic_factory.h"
```

Comment:
```suggestion
template<typename T>
class generic_factory;
```
There is no need to expose the whole header here.